packages/contracts/test/invariant/diamond/facets/PoolFacetHandler.sol

```diff
@@ -0,0 +1,266 @@
+     * It assumes the new price is within the specified range and updates the collateral ratio in the UbiquityPoolFacet.
+     * @param _newPrice The new price of the stable USD token, scaled by 1e8 (e.g., a price of $1 is represented as 1e8).
+     */
+    function setStableUsdPrice(uint256 _newPrice) public {
+        vm.assume(_newPrice >= 0.5e8 && _newPrice <= 1.5e8); // Assume a range for testing
+
+        stableUsdPriceFeed.updateMockParams(
+            1, // round id
+            int256(_newPrice),
+            block.timestamp, // started at
+            block.timestamp, // updated at
+            1 // answered in round
+        );
+
+        uint256 newCollateralRatio = uint256(1e6 * 1e8).div(_newPrice);
+        ubiquityPoolFacet.setCollateralRatio(newCollateralRatio);
+    }
+
+    /**
+     * @notice Manipulates the ETH/USD price using a mock ChainLink price feed.
+     * @dev This function assumes the new price is within the specified range and updates the ETH/USD price
+     * in the corresponding mock price feed.
+     * @param _newPrice The new ETH/USD price, scaled by 1e8 (e.g., a price of $1,000 is represented as 1000e8).
+     */
+    function setEthUsdPrice(uint256 _newPrice) public {
+        vm.assume(_newPrice >= 1000e8 && _newPrice <= 5000e8);
+
+        ethUsdPriceFeed.updateMockParams(
+            1, // round id
+            int256(_newPrice), // new price
+            block.timestamp, // started at
+            block.timestamp, // updated at
+            1 // answered in round
+        );
+    }
+
+    /**
+     * @notice Mints Ubiquity Dollar tokens using specified collateral and governance token inputs.
+     * @dev Assumes that the dollar amount is within a safe range, the minimum dollar output is less than or equal to the input amount,
+     * and the collateral and governance inputs are valid. The function then pranks the specified user to simulate a mint transaction.
+     * @param _dollarAmount The amount of Ubiquity Dollars to mint.
+     * @param _dollarOutMin The minimum amount of Ubiquity Dollars expected to be received from the minting process.
+     * @param _maxCollateralIn The maximum amount of collateral tokens to use for minting.
+     * @param _maxGovernanceIn The maximum amount of governance tokens to use for minting.
+     * @param _isOneToOne A boolean flag indicating whether the minting process should be executed on a one-to-one basis with the collateral.
+     */
+    function mintUbiquityDollars(
+        uint256 _dollarAmount,
+        uint256 _dollarOutMin,
+        uint256 _maxCollateralIn,
+        uint256 _maxGovernanceIn,
+        bool _isOneToOne
+    ) public {
+        vm.assume(
+            _dollarAmount > 0 && _dollarAmount < type(uint256).max.div(2)
+        );
+        vm.assume(_dollarOutMin <= _dollarAmount);
+        vm.assume(
+            _maxCollateralIn > 0 && _maxCollateralIn < type(uint256).max.div(2)
+        );
+        vm.assume(
```

Comment:
Why do we `div(2)` here? What's the difference between `type(uint256).max.div(2)` and `type(uint256).max` in this particular case?